feat: allow override of volume and volumemount

H777K opened the pull request:
This PR ensures, that the default volume and volumemount wont be created, if they are defined in podtemplate to e.g. use an PVC instead of an emptydir

bavarianbidi commented on internal/spec/spec.go:
```diff
@@ -181,30 +187,56 @@ func ToValidLabel(input string) string {
+			for _, volMounts := range container.VolumeMounts {
+				// Volumemount paths e.g. /runner and /runner/ are threated equal
+				// The last one in the pod spec will take precedence, which can lead to unexpected behavior
+				if volMounts.MountPath == runnerVolumeMountPath || volMounts.MountPath == runnerVolumeMountPath+"/" {
```
```suggestion
				if volMounts.MountPath == filepath.Clean(runnerVolumeMountPath) {
```

by using the `path/filepath` pkg from the standard lib (https://pkg.go.dev/path/filepath#Clean)

bavarianbidi replied:
example snippet: https://go.dev/play/p/nR8FvWytZRD

bavarianbidi commented on internal/spec/spec.go:
```diff
@@ -15,6 +15,12 @@ import (
 )
 
+var (
+	runnerVolumeName      = "runner"
+	runnerVolumeMountPath = "/runner"
```
why not moving these two vars into the `const` section in `internal/provider/provider.go`

H777K replied:
since the three variables `runnerVolumeName`, `runnerVolumeMountPath` and `runnerVolumeEmptyDir` are only used in the `internal/spec/spec.go` pkg, i wanted to keep them there.. it could also be confusing if i split these variables between the two pkgs

bavarianbidi reviewed (commented):
can you add a unit test for at least the new introduced func `CreateRunnerVolumeMount`? And as the `CreateRunnerVolume` func has also changed, a small unit test might be worth the effort :pray:

H777K commented:
> can you add a unit test for at least the new introduced func `CreateRunnerVolumeMount`? And as the `CreateRunnerVolume` func has also changed, a small unit test might be worth the effort 🙏

i have added two tests in `internal/provider/provider_test.go`
